dipy/direction/probabilistic_direction_getter.pyx

```diff
@@ -1,62 +1,34 @@
+# cython: boundscheck=False
+# cython: initializedcheck=False
+# cython: wraparound=False
+
 """
 Implementation of a probabilistic direction getter based on sampling from
-discrete distribution (pmf) at each step of the tracking."""
-import numpy as np
-from dipy.direction.peaks import peak_directions, default_sphere
-from dipy.reconst.shm import order_from_ncoef, sph_harm_lookup
-from dipy.tracking.local.direction_getter import DirectionGetter
-from dipy.tracking.local.interpolation import trilinear_interpolate4d
-
-
-def _asarray(cython_memview):
-    # TODO: figure out the best way to get an array from a memory view.
-    # `np.array(view)` works, but is quite slow. Views are also "array_like",
-    # but using them as arrays seems to also be quite slow.
-    return np.fromiter(cython_memview, float)
-
-
-class PmfGen(object):
-    pass
-
-
-class SimplePmfGen(PmfGen):
-
-    def __init__(self, pmf_array):
-        if pmf_array.min() < 0:
-            raise ValueError("pmf should not have negative values")
-        self.pmf_array = pmf_array
-
-    def get_pmf(self, point):
-        return trilinear_interpolate4d(self.pmf_array, point)
-
+discrete distribution (pmf) at each step of the tracking.
+"""
 
-class SHCoeffPmfGen(PmfGen):
+from random import random
 
-    def __init__(self, shcoeff, sphere, basis_type):
-        self.shcoeff = shcoeff
-        self.sphere = sphere
-        sh_order = order_from_ncoef(shcoeff.shape[-1])
-        try:
-            basis = sph_harm_lookup[basis_type]
-        except KeyError:
-            raise ValueError("%s is not a known basis type." % basis_type)
-        self._B, m, n = basis(sh_order, sphere.theta, sphere.phi)
+import numpy as np
+cimport numpy as np
```

Comment:
it can be confusing. Can you use `cnp` as an alias?

Reply:
Yes, I can, but I'll wait for another request because there's no name collision and imo it's not confusing. The `cimport` version is **only** used for the variable type, so in `cdef` and argument definition. For example `np.float_t` clearly use the `cimport`, while `np.dot` doesn't.

As I said, I'll change it to `cnp` if it cpnfuse the team!